packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -312,6 +308,15 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
     function speedUpRelay(DepositData memory _depositData) public {
         bytes32 depositHash = _getDepositHash(_depositData);
         RelayData storage relay = relays[depositHash];
+        require(
+            _getOptimisticOracle().getState(
```

Comment:
Might be a dumb question, but why do we need to check this? Don't we throw out a request if it's disputed, but _save_ the fast relay? So theoretically (probably would never happen) a fast relayed could speed up a relay that's disputed, and that fast relay would apply whenever it gets re-proposed? Basically just wondering why we need to check this if the fast relater is responsible for checking the request validity anyway.

Not saying we should change this now -- just that we might want to discuss this more.

Reply:
>  So theoretically (probably would never happen) a fast relayed could speed up a relay that's disputed, and that fast relay would apply whenever it gets re-proposed?

This is a good point. I'll think on this I think you. might be on to something,

Reply:
I guess you could think of this line as protecting the fast relayer in the case where the (invalid) relay has an incorrect fast relay fee % and the fast relayer would receive an incorrect amount of fees. This protection might not be necceessary.

Reply:
I ended up removing the check for whether a relay is disputed because I agree that the onus is on the caller to determine whether a relay should be sped up. If the relay is valid, then the `instantRelayer` address is carried over. If the relay is invalid, then the instant relayer loses their funds.